src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/services/CFTSensitiveTaskEventLogsDatabaseServiceTest.java

```diff
@@ -0,0 +1,94 @@
+import uk.gov.hmcts.reform.wataskmanagementapi.repository.SensitiveTaskEventLogsRepository;
+import uk.gov.hmcts.reform.wataskmanagementapi.utils.ServiceMocks;
+
+import java.time.OffsetDateTime;
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.ScheduledThreadPoolExecutor;
+
+import static java.util.Collections.singletonList;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static uk.gov.hmcts.reform.wataskmanagementapi.cft.enums.CFTTaskState.UNCONFIGURED;
+
+public class CFTSensitiveTaskEventLogsDatabaseServiceTest  extends SpringBootIntegrationBaseTest {
+
+    @Autowired
+    SensitiveTaskEventLogsRepository sensitiveTaskEventLogsRepository;
+
+    @MockBean
+    private IdamWebApi idamWebApi;
+    @MockBean
+    private ServiceAuthorisationApi serviceAuthorisationApi;
+    @MockBean
+    private CamundaServiceApi camundaServiceApi;
+    @MockBean
+    private RoleAssignmentServiceApi roleAssignmentServiceApi;
+    @MockBean
+    CFTTaskDatabaseService cftTaskDatabaseService;
+
+    private ServiceMocks mockServices;
+
+    CFTSensitiveTaskEventLogsDatabaseService cftSensitiveTaskEventLogsDatabaseService;
+
+    @BeforeEach
+    void setUp() {
+        cftSensitiveTaskEventLogsDatabaseService =
+            new CFTSensitiveTaskEventLogsDatabaseService(sensitiveTaskEventLogsRepository, cftTaskDatabaseService);
+
+        mockServices = new ServiceMocks(
+            idamWebApi,
+            serviceAuthorisationApi,
+            camundaServiceApi,
+            roleAssignmentServiceApi
+        );
+    }
+
+    @Test
+    void should_process_and_save_sensitive_task_event_log() {
+        String taskId = UUID.randomUUID().toString();
+        String caseId = "Some caseId";
+        TaskResource taskResource = new TaskResource(
+            taskId,
+            "someTaskName",
+            "someTaskType",
+            UNCONFIGURED,
+            OffsetDateTime.parse("2022-05-09T20:15:45.345875+01:00")
+        );
+        taskResource.setCreated(OffsetDateTime.now());
+        taskResource.setCaseId(caseId);
+
+        final List<String> roleNames = singletonList("tribunal-caseworker");
+
+        List<RoleAssignment> roleAssignments =
+            mockServices.createTestRoleAssignments(roleNames);
+
+        when(cftTaskDatabaseService.findByIdOnly(taskId)).thenReturn(java.util.Optional.of(taskResource));
+
+        cftSensitiveTaskEventLogsDatabaseService.processSensitiveTaskEventLog(taskId,
+            roleAssignments, ErrorMessages.ROLE_ASSIGNMENT_VERIFICATIONS_FAILED_ASSIGNEE);
+
+        ExecutorService executorService = new ScheduledThreadPoolExecutor(1);
+        executorService.execute(() -> {
+            verify(sensitiveTaskEventLogsRepository, times(1)).save(any(SensitiveTaskEventLog.class));
+        });
+    }
+}
```

Comment:
should we create a test that covers the AC specifically? I.e. when we call cancelTask() and the taskResourceQueryResult.isEmpty for example? that way we can verify/assert from the db that the record saved has the failure reason mapped in the message, user_data, task_data etc

Reply:
This scenario above is still not covered in this IT. We are mocking essentially all facets of the expectation and not actually querying the sensitive task and then asserting the message, user_data and task-data.